mlx/backend/metal/kernels/reduction/ops.h

```diff
@@ -5,9 +5,11 @@
 #include <metal_atomic>
 #include <metal_simdgroup>
 
+#ifndef MLX_METAL_JIT
```

Comment:
That's a bit annoying. I think we 'll waste a fair amount of time forgetting to do that. I don't have anything better to propose... just commenting.

Reply:
Yea it's annoying. Getting the includes right is annoying in general. There is probably a way to avoid this but it requires some care in what you include where and in what order (which is quite brittle).

Reply:
I was able to remove a few of these and I think I can remove the rest in #1132.